engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -58,17 +60,27 @@ public DockerOverseer(
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            this.containerId = containerId;
+            executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS);
+        });
         this.init = init;
     }
 
     @Override
     public CompletableFuture<ExecutionOutcome> abort() {
-        return result.isDone() ? result : CompletableFuture.completedFuture(
-                new ExecutionOutcome(Stage.ABORT, "Abort is not yet supported for docker analysis", null)
-        );
+        String logs = getExecutionLogs();
+        return CompletableFuture.supplyAsync(() -> {
+            if (result.isDone()) {
+                return result.join();
+            } else {
+                client.stopContainerCmd(containerId).exec();
```

Comment:
According to the documentation, there are dedicated conditions for stopContainerCmd:
```
NotFoundException - No such container
NotModifiedException - Container already stopped
```
There error conditions (well, at least one of them) should be handled explicitly as idempotent success.
Yes, this is tricky as completion of a container can race with completion of the `result` future and the code has to account for that.

Reply:
Thanks. For this purpose, we will not be using generic exception then and will use specific.